File: core/src/com/broll/gainea/server/core/cards/impl/play/C_Vagrant.java
package com.broll.gainea.server.core.cards.impl.play;

import com.broll.gainea.server.core.cards.AbstractCard;
import com.broll.gainea.server.core.map.Area;
import com.broll.gainea.server.core.map.AreaType;
import com.broll.gainea.server.core.map.Location;
import com.broll.gainea.server.core.objects.Monster;
import com.broll.gainea.server.core.objects.Soldier;
import com.broll.gainea.server.core.utils.LocationUtils;
import com.broll.gainea.server.core.utils.UnitControl;

import java.util.List;
import java.util.stream.Collectors;

public class C_Vagrant extends AbstractCard {
    public C_Vagrant() {
        super(0, "Einsamer Landstreicher", "Platziert einen neutralen Vagabund (2/2) auf ein beliebiges freies Feld.");
    }

    @Override
    public boolean isPlayable() {
        return true;
    }

    @Override
    protected void play() {
        Soldier soldier = new Soldier(null);
        soldier.setIcon(1);
        soldier.setName("Vagabund");
        soldier.setStats(2,2);
        List<Area> locations = game.getMap().getAllAreas().stream().filter(Location::isFree).collect(Collectors.toList());
        Location location = selectHandler.selectLocation("Standort für Vagabund festlegen", locations);
        UnitControl.spawn(game, soldier, location);
    }


}

File: server/src/main/com/broll/gainea/server/core/cards/impl/play/C_Vagrant.kt
package com.broll.gainea.server.core.cards.impl.play

import com.broll.gainea.server.core.cards.Card
import com.broll.gainea.server.core.objects.Soldier
import com.broll.gainea.server.core.utils.UnitControl.spawn

class C_Vagrant : Card(0, "Einsamer Landstreicher", "Platziert einen neutralen Vagabund (2/2) auf ein beliebiges freies Feld.") {
    override val isPlayable: Boolean
        get() = true

    override fun play() {
        val soldier = Soldier(game.neutralPlayer)
        soldier.icon = 1
        soldier.name = "Vagabund"
        soldier.setStats(2, 2)
        val locations = game.map.allAreas.filter { it.isFree }
        val location = selectHandler.selectLocation("Standort für Vagabund festlegen", locations)
        spawn(game, soldier, location)
    }
}
